### ISSUE TITLE
Should not forward local proxy settings to remote ssh server

### ISSUE THREAD

**Issue by kigawas** [microsoft/vscode-remote-release#957]
TL;DR: You can work it around by `code --no-proxy-server`

- VSCode Version: 1.36.1
- Local OS Version: Centos 7.5
- Remote OS Version: Ubuntu 18.04
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. Set local proxy in ENV (`export http_proxy`)
2. Install [crates](https://marketplace.visualstudio.com/items?itemName=serayuzgur.crates) extension
3. Open any Cargo.toml file in a __remote__ rust project
4. Crates will print errors in console (`Developer: Toggle developer tools`)

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

**Comment by kigawas** [comment]
FYI
```
console.ts:134 [Extension Host] RequestError: Error: getaddrinfo ENOTFOUND proxy.example.com proxy.example.com:8080
	at new RequestError (/home/kigawas/.vscode-server/extensions/serayuzgur.crates-0.4.3/node_modules/request-promise-core/lib/errors.js:14:15)
	at Request.plumbing.callback (/home/kigawas/.vscode-server/extensions/serayuzgur.crates-0.4.3/node_modules/request-promise-core/lib/plumbing.js:87:29)
	at Request.RP$callback [as _callback] (/home/kigawas/.vscode-server/extensions/serayuzgur.crates-0.4.3/node_modules/request-promise-core/lib/plumbing.js:46:31)
	at self.callback (/home/kigawas/.vscode-server/extensions/serayuzgur.crates-0.4.3/node_modules/request/request.js:185:22)
	at Request.emit (events.js:182:13)
	at Request.onRequestError (/home/kigawas/.vscode-server/extensions/serayuzgur.crates-0.4.3/node_modules/request/request.js:881:8)
	at ClientRequest.emit (events.js:187:15)
	at onerror (/home/kigawas/.vscode-server/bin/2213894…/node_modules/agent-base/index.js:100:9)
	at callbackError (/home/kigawas/.vscode-server/bin/2213894…/node_modules/agent-base/index.js:122:5)
	at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by roblourens** [comment]
I don't quite understand what's going on, is crates trying to access a local server and not seeing no_proxy? Otherwise why shouldn't it get the proxy settings?

**Comment by kigawas** [comment]
Crates use the proxy server set locally, but remote server cannot get access to the proxy server.

**Comment by roblourens** [comment]
I'm not sure how that's possible. Are those set as environment variables? If you enter `echo $https_proxy` in a remote vscode terminal do you see those values? I want to make sure that they are not actually in the remote environment.

**Comment by kigawas** [comment]
Yes. No.

**Comment by roblourens** [comment]
Hm. They must still be set in your remote environment anyway. vscode does not copy them over. Otherwise this is an issue for the crates extension which seems unlikely too.

**Comment by kigawas** [comment]
@roblourens Not possible since remote environment has nothing to do with local. And crates extension has no reason dealing with environment, not sure but perhaps it's related with `request-promise` package used by crates.

**Comment by kigawas** [comment]
@roblourens Not only crates, I can also reproduce this bug on `vscode-lldb` when downloading lldb binaries. You can check their code [here](https://github.com/vadimcn/vscode-lldb/blob/master/extension/install.ts).

For reproducing the bug, check the instruction [here](https://github.com/vadimcn/vscode-lldb/wiki/Installing-LLDB) and start debugging, it will ask you to download package, then it stuck.

![image](https://user-images.githubusercontent.com/4182346/61511181-e0e88e00-aa30-11e9-93cc-1f97b077c6f4.png)

**Comment by kigawas** [comment]
@roblourens 
And the same phenomenon happened on vscode-lldb can also be worked around by `code --no-proxy-server`.

**Comment by roblourens** [comment]
Oh, so did you set any vscode proxy settings like `http.proxy`?

**Comment by kigawas** [comment]
No, just environment variables

**Comment by roblourens** [comment]
Still no idea what the problem is. I don't think it's possible that vscode is copying these properties to the remote environment.
